src/eos-dropbox-app

```diff
@@ -121,8 +121,13 @@ class DropboxLauncher():
         self._open_dropbox_directory()
 
     def _name_lost(self, *args):
-        logging.info("Another instance of dropbox is already running")
-        self._already_running = True
+        if not get_dropbox_directory():
+            logging.info("Another instance of dropbox is already running but no "
+                         "Dropbox folder was found; launching the daemon again")
+            self._launch_dropbox_daemon()
+        else:
+            logging.info("Another instance of dropbox is already running")
+            self._already_running = True
         self._open_dropbox_directory()
```

Comment:
In any case, assuming that my previous comment is irrelevant and Dropbox deals fine with multiple instances running, this line should only be run `if not get_dropbox_directory()`.